libyul/AsmAnalysis.cpp

```diff
@@ -766,3 +772,20 @@ bool AsmAnalyzer::validateInstructions(FunctionCall const& _functionCall)
 {
 	return validateInstructions(_functionCall.functionName.name.str(), nativeLocationOf(_functionCall.functionName));
 }
+
+bool AsmAnalyzer::validateObjectStructure()
+{
+	if (m_eofVersion.has_value())
+	{
+		if (util::contains(m_objectName, '.')) // No dots in object name for EOF
+		{
+			m_errorReporter.syntaxError(9822_error, SourceLocation{}, // TODO: How to get proper location?
+				fmt::format("In EOF context object name \"{objectName}\" must not contain 'dot' character.",
+					fmt::arg("objectName", m_objectName)));
```

Comment:
> ```c++
> // TODO: How to get proper location?
> ```

You can't I guess. For some reason we only internally have ASTs for the `code` part of each object, not for the Yul file as a whole. So we don't know the location of the name and there's no AST node corresponding to it.

For now my suggestion would be to point at the root of the AST (and also check on the CLI if reporting this error this way makes sense visually; it will probably point at the opening brace?).

Reply:
Also, this is the way we wrap long calls:
```suggestion
			m_errorReporter.syntaxError(
				9822_error,
				SourceLocation{}, // TODO: How to get proper location?
				fmt::format(
					"In EOF context object name \"{objectName}\" must not contain 'dot' character.",
					fmt::arg("objectName", m_objectName)
				)
			);
```

Same in `analyzeStrictAssertCorrect()`, `StackCompressor.cpp`, `StackLimitEvader.cpp` and `Common.cpp`.